docs/extensions.md

```diff
@@ -0,0 +1,308 @@
+
+All you need to do is provide a js module exporting your `Component`, this component will be loaded at runtime by the Tekton Dashboard host and injected into the page.
+
+For this to work, you will need to create your component using the shared objects provided by the host, namely you won't `import React, { Component } from 'react';` but use `window.React` instead.
+
+To deploy the service based extension with frontend code run the following command:
+
+```bash
+cat <<EOF | kubectl apply -n tekton-pipelines -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: sample-extension
+  labels:
+    app: sample-extension
+    tekton-dashboard-extension: "true"
+  annotations:
+    tekton-dashboard-display-name: Hello
+    tekton-dashboard-endpoints: sample.bundle
+    tekton-dashboard-bundle-location: bundle
+spec:
+  ports:
+  - port: 3000
+    targetPort: 3000
+  selector:
+    app: sample-extension
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-extension
+  labels: 
+    app: sample-extension
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sample-extension
+  template:
+    metadata:
+      labels:
+        app: sample-extension
+    spec:
+      containers:
+      - name: node
+        image: node:latest
+        ports:
+        - containerPort: 3000
+        command:
+        - bash
+        args:
+        - -c
+        - |
+          cat <<EOF > frontend.js
+          const React = window.React;
+          class Extension extends React.Component {
+            state = {
+              message: 'Loading ...'
+            };
+            componentDidMount() {
+              fetch('/v1/extensions/sample-extension/sample')
+                .then(response => response.text())
+                .then(message => this.setState({ message }));
+            };
+            render() {
+              const { message } = this.state;
+              return React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("h1", null, message));
+            }
+          }
+          export default Extension;
+          EOF
+          cat <<EOF > server.js
+          const express = require('express');
+          const app = express();
+          app.get('/sample', (req, res) => res.send('Hello Tekton Dashboard !'));
+          app.get('/bundle', (req, res) => res.sendFile('/frontend.js'));
+          app.listen(3000, '0.0.0.0');
+          EOF
+          npm install express
+          node ./server.js
+EOF
+```
+
+You can see from the code above that the `Service` annotations changed:
+- `tekton-dashboard-endpoints: sample.bundle` means that both `sample` and `bundle` endpoints exist on the extension
+- `tekton-dashboard-bundle-location: bundle` means that the frontend code will be loaded at the `/bundle` path
+
+In the extension `Pod`, a new `frontend.js` file is generated containing the frontend code and the `/bundle` route is registered in the `express` server to serve the frontend js file.
+
```

Comment:
We should clarify that in a real extension the frontend code would be written and bundled separately, devs aren't expected to inline an ES module into the extension resource and are free to use additional libraries if they wish. The extension API only requires that they provide an ES module that exposes a React component for us to load.

Reply:
fixed